pin/pin.go

```diff
@@ -86,6 +87,11 @@ type Pinner interface {
 	Pin(context.Context, node.Node, bool) error
 	Unpin(context.Context, *cid.Cid, bool) error
 
+	// Update updates a recursive pin from one cid to another
+	// this is more efficient than simply pinning the new one and unpinning the
+	// old one
+	Update(context.Context, *cid.Cid, *cid.Cid, bool) error
```

Comment:
I would name the parameters here.  The use of the bool in Update differs from the use in pin and unpin.